From cc74b0273f423a2857f8ba9d7b8c019020bd2feb Mon Sep 17 00:00:00 2001
From: Paul Licameli <paul.licameli@audacityteam.org>
Date: Wed, 3 Oct 2018 14:24:20 -0400
Subject: [PATCH] Fix return of reference to temporary object (that's bad!) ...

... This might be the reason for Linux crashes when playing-at-speed.

This was a bad thing to do with any compiler -- not to be blamed on
possible compiler bugs.

There was a warning about this in the XCode build.
---
 src/tracks/ui/Scrubbing.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/tracks/ui/Scrubbing.cpp b/src/tracks/ui/Scrubbing.cpp

index 2d3ed6cbd8..4332577bbc 100644
========================================================================
--- a/src/tracks/ui/Scrubbing.cpp	
+++ b/src/tracks/ui/Scrubbing.cpp	
@@ -1104,7 +1104,8 @@ const wxString &Scrubber::GetUntranslate
    static wxString empty;
 
    if (IsSpeedPlaying()) {
-      return XO("Playing at Speed");
+      static wxString result = XO("Playing at Speed");
+      return result;
    }
    else if (HasMark()) {
       auto &item = FindMenuItem(Seeks() || TemporarilySeeks());

